File: Python/beadsort/mistral-saba-24b.py

import random

def beadsort(arr):
    n = len(arr)
    # Create a 2D list to represent the beads
    beads = [['.' for _ in range(n)] for _ in range(max(arr))]

    # Place beads in the 2D list
    for i in range(n):
        for j in range(arr[i]):
            beads[j][i] = 'O'

    # Sort the beads
    for i in range(n):
        count = 0
        for j in range(n):
            if beads[j][i] == 'O':
                count += 1
        for j in range(n):
            if j < count:
                beads[j][i] = 'O'
            else:
                beads[j][i] = '.'

    # Extract the sorted array
    sorted_arr = []
    for i in range(n):
        count = 0
        for j in range(n):
            if beads[j][i] == 'O':
                count += 1
        sorted_arr.append(count)

    return sorted_arr

# Generate 1000 random inputs
random_inputs = [random.randint(0, 100) for _ in range(1000)]

# Sort the random inputs using beadsort
sorted_inputs = beadsort(random_inputs)

File: Java/beadsort/MistralSaba24b.java

import java.util.Arrays;
import java.util.Random;

public class MistralSaba24b{

    public static void beadsort(int[] array) {
        int n = array.length;
        int max = Arrays.stream(array).max().getAsInt();

        // Create a 2D array to represent the beads
        boolean[][] beads = new boolean[max + 1][n];

        // Initialize the beads array
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < array[i]; j++) {
                beads[j][i] = true;
            }
        }

        // Perform the beadsort
        for (int i = 0; i < max; i++) {
            for (int j = 0; j < n; j++) {
                if (beads[i][j]) {
                    int k = j;
                    while (k > 0 && beads[i][k - 1]) {
                        // Swap beads
                        beads[i][k] = false;
                        beads[i][k - 1] = true;
                        k--;
                    }
                }
            }
        }

        // Extract the sorted array from the beads array
        for (int i = 0; i < n; i++) {
            for (int j = max; j >= 0; j--) {
                if (beads[j][i]) {
                    array[i] = j;
                    break;
                }
            }
        }
    }

    public static void main(String[] args) {
        Random random = new Random();
        int[] array = new int[1000];
        for (int i = 0; i < 1000; i++) {
            array[i] = random.nextInt(1000);
        }

        beadsort(array);
    }
}